regression-test/suites/nereids_syntax_p0/set_operation.groovy

```diff
@@ -242,4 +241,9 @@ suite("test_nereids_set_operation") {
     INTERSECT
     SELECT k1 FROM setOperationTable WHERE k2 > 0)
     """
+
+    order_qt_select43 """
+        SELECT * FROM (select k1, k3 from setOperationTableNotNullable order by k3 union all 
+            select k1, k5 from setOperationTable) t;
+    """
```

Comment:
add more case, to test all situation of legacy planner's syntax. such as set operation, w/ and w/o parenthesis, w/ and w/o cte and w/ and w/o query organization

Reply:
done